assets/src/edit-story/components/form/color.js

```diff
@@ -18,107 +18,184 @@
+  visibility: ${({ isVisible }) => (isVisible ? 'visible' : 'hidden')};
 `;
 
-const Container = styled.div`
-  color: ${({ theme }) => theme.colors.fg.v1};
-  font-family: ${({ theme }) => theme.fonts.body2.family};
-  font-size: ${({ theme }) => theme.fonts.body2.size};
-  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
-  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
-  display: flex;
-  flex-direction: row;
-  justify-content: flex-start;
-  align-items: center;
-  margin-left: 5px;
-  margin-right: 5px;
+const transparentStyle = {
+  backgroundImage:
+    'conic-gradient(#fff 0.25turn, #d3d4d4 0turn 0.5turn, #fff 0turn .75turn, #d3d4d4 0turn 1turn)',
+  backgroundSize: '66.67% 66.67%',
+};
 
-  & > span {
-    margin-left: 12px;
+function printRGB(r, g, b) {
+  const hex = (v) => v.toString(16).padStart(2, '0');
+  return `${hex(r)}${hex(g)}${hex(b)}`.toUpperCase();
+}
+
+function getPreviewStyle(pattern, defaultColor) {
+  if (!pattern) {
+    if (!defaultColor) {
+      return transparentStyle;
+    }
+    return { backgroundColor: `#${defaultColor}` };
   }
-`;
+  const isSolidPattern = pattern.type === 'solid' || !pattern.type;
+  if (!isSolidPattern) {
+    return generatePatternCSS(pattern);
+  }
+  const {
+    color: { r, g, b, a },
+  } = pattern;
+  // If opacity is 0, create as transparent:
+  if (a === 0) {
+    return transparentStyle;
+  }
+
+  // Otherwise create color, but with full opacity
+  return generatePatternCSS({ color: { r, g, b } });
+}
+
+function getPreviewOpacity(pattern, specifiedOpacity = 1) {
+  if (!pattern) {
+    return specifiedOpacity * 100;
+  }
+  const isSolidPattern = pattern.type === 'solid' || !pattern.type;
+  if (!isSolidPattern) {
+    return specifiedOpacity * 100;
+  }
+  const {
+    color: { a = 1 },
+  } = pattern;
+  return a * 100;
+}
+
+function getPreviewText(pattern) {
+  if (!pattern) {
+    return null;
+  }
+  switch (pattern.type) {
+    case 'radial':
+      return __('Radial', 'web-stories');
+    case 'conic':
+      return __('Conic', 'web-stories');
+    case 'linear':
+      return __('Linear', 'web-stories');
+    case 'solid':
+    default:
+      const {
+        color: { r, g, b, a },
+      } = pattern;
+      if (a === 0) {
+        return null;
+      }
+      return printRGB(r, g, b);
+  }
+}
 
-function ColorInput({
-  onBlur,
-  onChange,
-  isMultiple,
-  opacity,
-  label,
-  value,
-  ...rest
-}) {
-  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
+function ColorInput({ onChange, isMultiple, opacity, value, label }) {
+  const previewStyle = getPreviewStyle(isMultiple ? null : value);
+  const previewText = getPreviewText(value);
+  const previewOpacity = getPreviewOpacity(value, opacity);
+
+  const {
+    actions: { showColorPickerAt, hideColorPicker },
+  } = useColorPicker();
+
+  const ref = useRef();
+
+  const handleOpenEditing = useCallback(() => {
+    showColorPickerAt(ref.current, {
```

Comment:
Per above, we need a way to hide gradient options in the color picker, for the text color at least.

Reply:
Done in 7e104a7e.